sky/backends/backend_utils.py

```diff
@@ -173,6 +173,16 @@
     ('available_node_types', 'ray.head.default', 'node_config',
      'azure_arm_parameters', 'cloudInitSetupCommands'),
 ]
+# These keys are expected to change when provisioning on an existing cluster,
+# but they don't actually represent a change that requires re-provisioning the
+# cluster.  If the cluster yaml is the same except for these keys, we can safely
+# skip reprovisioning. See _deterministic_cluster_yaml_hash.
+_RAY_YAML_KEYS_TO_REMOVE_FOR_HASH = [
+    # On first launch, availability_zones will include all possible zones. Once
+    # the cluster exists, it will only include the zone that the cluster is
+    # actually in.
+    ('provider', 'availability_zone'),
```

Comment:
Azure may have a different name for zone. May need to double check.

Reply:
Azure doesn't use it at all https://github.com/skypilot-org/skypilot/blob/346646949beead6d93f793851e959da4b56246b0/sky/clouds/azure.py#L418-L419